src/engine/srv_internal.h

```diff
@@ -181,6 +195,9 @@ sched_create_task(struct dss_xstream *dx, void (*func)(void *), void *arg,
 	struct sched_info	*info = &dx->dx_sched_info;
 	int			 rc;
 
+	if (sched_xstream_stopping())
+		return -DER_SHUTDOWN;
```

Comment:
you can use si_stop here? or I miss sth?

Reply:
This is to check current xstream stopping, and it's guaranteed that all xstreams's stopping are set before any dx_shutdown is set.

Reply:
Sorry, I still do not understand here. if wait_all_exited does not make sure all running ULT are exited, then how does this help here?

As I understand this will only stop the xstream to create new task/thread.  how about those inflight ULT?  If I understand correctly, then this is no better than a flag? i.e. it does not need future(lock) here?   It looks like DAOS has to track all ULTs?  Do I miss sth?